dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/query/SolrQuerySanitizer.java

```diff
@@ -43,6 +43,46 @@ public String sanitizeQuery(String query, Collection<DatasetFieldType> datasetFi
         return query;
     }
 
+    public String sanitizeRorQuery(String query) {
+        StringBuilder transformedQuery = new StringBuilder();
+
+        boolean currentCharIsEscaped = false;
+        boolean insideQuotation = false;
```

Comment:
IMO there is no need to keep track of quotation. I don't think we need to support such queries. Just escaping every special character should be fine:
```
public String sanitizeRorQuery(String query) {
StringBuilder transformedQuery = new StringBuilder();
for (int i=0; i<query.length(); ++i) {
if (SOLR_SPECIAL_CHARACTERS.contains(currentChar)) {
  transformedQuery.append("\\" + currentChar);
} else {
  transformedQuery.append(currentChar);
}
return transformedQuery.toString();
}
```